components/scream/src/physics/rrtmgp/atmosphere_radiation.cpp

```diff
@@ -136,39 +155,38 @@ void RRTMGPRadiation::init_buffers(const ATMBufferManager &buffer_manager)
   m_buffer.sfc_alb_dif = decltype(m_buffer.sfc_alb_dif)("surf_alb_diffuse", mem, m_ncol, m_nswbands);
   mem += m_buffer.sfc_alb_dif.totElems();
 
-  // 3d array
-  m_buffer.gas_vmr = decltype(m_buffer.gas_vmr)("gas_vmr", mem, m_ncol, m_nlay, m_ngas);
-  mem += m_buffer.gas_vmr.totElems();
-
   int used_mem = (reinterpret_cast<Real*>(mem) - buffer_manager.get_memory())*sizeof(Real);
   EKAT_REQUIRE_MSG(used_mem==requested_buffer_size_in_bytes(), "Error! Used memory != requested memory for RRTMGPRadiation.");
 } // RRTMGPRadiation::init_buffers
 
 void RRTMGPRadiation::initialize_impl(const util::TimeStamp& /* t0 */) {
   // Names of active gases
-  // TODO: this needs to be not hard-coded...I wanted to get these from
-  // input files, but need to get around the rrtmgp_initializer logic.
-  // Maybe can store the gas names somewhere else?
+  // TODO: Is there any way to store the gas_names_1d information somewhere once,
+  // rather than populate this array every step?
+  // It appears that "gas_concs.init(...)" requires a YAKL string1d array as input,
+  // which would need to be defined with its size pre-initialization, but this breaks the
+  // flexibility to define the full set of gases at runtime.
+  // So for now we create the string1d gas_names_1d on the fly to pass to gas_concs.init. 
```

Comment:
This is what the comment refers too.  My understanding is that `gas_names_1d` is a YAKL structure and that it needs to have the size known at construction.  But we want the list of active gases to be a runtime option, i.e. m_ngas not known at compile time.  Since `gas_names_1d` is a required input to initialize `gas_concs` I don't see any way around using the YAKL structure here, like for example using a `std::vector` or `std::set` instead, which would allow us to increase it's size based on a runtime list.  @brhillman , any thoughts?

Reply:
I'm not following. I'm saying, you could have store the string1d array in the class, and fill it just once. The comment says we're populating the array at every step, I was simply suggesting to store it. The fact that it is a YAKL array vs a std vector is irrelevant in this context.

Reply:
I may be misunderstanding how the YAKL array can be stored in the class.  From what I can see, when I declare a `string_1d` array in say the `atmosphere_radiation.hpp` file I need to state the size of the array.  Much like if I wanted to do something like `std::string a[10] = {blah,blah,blah...}`.  In this case I don't know that since m_ngas is set by the `input.yaml` file.

In contrast, the `std::vector` has at least the `push_back` option which would allow me to increase the size after `input.yaml` has been read in.  But maybe there is a way to declare an arbitrary `string_1d` YAKL array and then set the size at runtime?  I'll have to take a look.

Reply:
You can declare an arbitrary `string_1d` and allocate it later. That would look like
```
string1d mystr;
mystr = string1d("mystr", nstr);
```
I'm not 100% sure that will work if it's a class variable, but maybe? But also, I'm not sure why we are worried about it here, this is initialization code that's not called every step. The string data from the `gas_concs` object is being stored in the rrtmgp namespace I believe, and this is really just to populate that.

Reply:
It looks like gas_concs also gets initialized every run step.  Is this not intentional?

Reply:
see: https://github.com/E3SM-Project/scream/blob/f1f9107dd74fa919e35d8c160ea72438d1749722/components/scream/src/physics/rrtmgp/atmosphere_radiation.cpp#L268

Reply:
YAKL arrays are not static arrays. You can size them later, just like vectors. The fact that the var is a member of the class does not matter.